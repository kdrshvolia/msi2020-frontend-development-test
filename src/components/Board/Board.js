import React from 'react';
import PropTypes from 'prop-types';
import SearchJokeForm from '../SearchJokeForm/SearchJokeForm';
import JokeContainer from '../JokeContainer/JokeContainer';
import JokesList from '../JokesList/JokesList';
import Header from '../Header/Header';

const Board = ({
  setInputText,
  options,
  setType,
  setCategory,
  chosenOption,
  chosenCategory,
  getJoke,
  categories,
  jokes,
  addToFav
}) => {
  return (
    <>
      <Header title="Hey!" subtitle="Letâ€™s try to find a joke for you:" />
      <SearchJokeForm
        setInputText={setInputText}
        options={options}
        setType={setType}
        chosenOption={chosenOption}
        chosenCategory={chosenCategory}
        getJoke={getJoke}
        jokes={jokes}
        categories={categories}
        setCategory={setCategory}
      />
      {jokes.length === 0 ? null : <JokesList jokesList={jokes} addToFav={addToFav} />}
    </>
  );
};

Board.propTypes = {
  options: PropTypes.shape({
    random: PropTypes.string.isRequired,
    categories: PropTypes.string.isRequired,
    search: PropTypes.string.isRequired,
  }).isRequired,
  setType: PropTypes.func.isRequired,
  setCategory: PropTypes.func.isRequired,
  chosenOption: PropTypes.string.isRequired,
  chosenCategory: PropTypes.string,
  setInputText: PropTypes.func.isRequired,
  getJoke: PropTypes.func.isRequired,
  jokes: PropTypes.string.isRequired,
};

export default Board;
