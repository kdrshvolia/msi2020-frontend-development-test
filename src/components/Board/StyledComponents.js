import styled, { css } from 'styled-components';const BoardWrapper = styled.div`  display: flex;  max-width: 100%;  flex-direction: row;  justify-content: center;  align-items: stretch;  flex: 2;  overflow-y: auto;  overflow-x: hidden;  min-height: 100vh;  position: relative;`;const Favorites = styled.div`  display: flex;  flex-direction: column;  flex-basis: 40%;  background: #f8f8f8;  box-sizing: border-box;  padding: 40px;  @media (max-width: 992px) {    display: none;  }  &:last-of-type {    display: none;    background: #ffffff;    @media (max-width: 992px) {      background: ${(props) => (props.isSideBarOpened ? '#f8f8f8' : '#ffffff')};      padding: 80px 40px 40px;      box-sixing: border-box;      display: block;      position:  absolute;      transform: ${(props) => (props.isSideBarOpened ? 'translateX(-100%)' : 'none')};       top: 0;      width: 60%;      left: 100%;      transition: transform 0.5s ease;      min-height: 100vh;    }    @media (max-width: 600px) {      width: 100%;    }`;const Search = styled.div`  @media only screen and (min-width: 1200px) {    flex-basis: 60%;  }  @media only screen and (max-width: 1200px) {    flex-basis: 100%;  }  padding: 40px 7% 0;  position: relative;`;const FavoritesBtn = styled.div`  display: none;  @media (max-width: 992px) {    display: block;  }`;const FavoritesList = styled.div`  @media (max-width: 992px) {    display: ${(props) => (props.isSideBarOpened ? 'flex' : 'none')};    z-index: 2;  }`;const SideBarOverlay = styled.div`  position: fixed;  background: rgba(0, 0, 0, 0.3);  height: 100%;  top: 0;  left: 0;  width: 100%;  display: ${(props) => (props.isSideBarOpened ? 'block' : 'none')};   @media (min-width: 992px) {    display: none;   }      @media (max-width: 600px) {    display: none;   }  }`;export { BoardWrapper, Favorites, Search, FavoritesList, FavoritesBtn, SideBarOverlay };