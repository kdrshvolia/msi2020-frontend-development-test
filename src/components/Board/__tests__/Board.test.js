import renderer from 'react-test-renderer';
import React from 'react';
import Board from '../Board';

describe('Board', () => {
  it('Renders correctly', () => {
    const toggleFav = jest.fn();
    const setCategory = jest.fn();
    const getJoke = jest.fn();
    const favoritedJokes = {};
    const options = { a: 'a', b: 'b' };
    const jokes = [];
    const categories = ['a', 'b', 'c'];
    const rendered = renderer.create(
      <Board
        toggleFav={toggleFav}
        favoritedJokes={favoritedJokes}
        setCategory={setCategory}
        getJoke={getJoke}
        jokes={jokes}
        options={options}
        categories={categories}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
