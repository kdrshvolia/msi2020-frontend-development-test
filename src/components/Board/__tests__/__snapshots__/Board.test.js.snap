// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Renders correctly 1`] = `
<div
  className="sc-fzqBZW fhiFbC"
>
  <div
    className="sc-fzoyAV bYWzIm"
  >
    <h3
      className="sc-fznxsB eImrbF"
    >
      MSI 2020
    </h3>
    <div>
      <h1>
        Hey!
      </h1>
      <h2>
        Letâ€™s try to find a joke for you:
      </h2>
    </div>
    <div>
      <div
        className="sc-AxjAm dWVCDA"
      >
        <input
          checked={true}
          className="sc-AxirZ hxQKcv"
          id="Random"
          name="search-type"
          onChange={[Function]}
          type="radio"
        />
         
        <label
          htmlFor="Random"
        >
          Random
        </label>
      </div>
      <div
        className="sc-AxjAm dWVCDA"
      >
        <input
          checked={true}
          className="sc-AxirZ hxQKcv"
          id="From categories"
          name="search-type"
          onChange={[Function]}
          type="radio"
        />
         
        <label
          htmlFor="From categories"
        >
          From categories
        </label>
      </div>
      <div
        className="sc-AxhCb gGkwKU"
      >
        <div
          className="sc-AxhUy jPMDTT"
          onClick={[Function]}
        >
          A
        </div>
        <div
          className="sc-AxhUy jPMDTT"
          onClick={[Function]}
        >
          B
        </div>
        <div
          className="sc-AxhUy jPMDTT"
          onClick={[Function]}
        >
          C
        </div>
      </div>
      <div
        className="sc-AxjAm dWVCDA"
      >
        <input
          checked={true}
          className="sc-AxirZ hxQKcv"
          id="Search"
          name="search-type"
          onChange={[Function]}
          type="radio"
        />
         
        <label
          htmlFor="Search"
        >
          Search
        </label>
      </div>
      <div
        className="sc-AxgMl hTOpWV"
      >
        <input
          className="sc-AxheI EDBAb"
          onChange={[Function]}
          placeholder="Free text search..."
          type="text"
        />
      </div>
      <div>
        <button
          className="sc-AxiKw bNMzOj"
          onClick={[Function]}
          type="button"
        >
          Get a joke
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-fzpjYC ePBISF"
  />
  <div
    className="sc-fzqNJr QiOxu"
  >
    <h3
      className="sc-fznJRM fAObgi"
    >
      Favourite
    </h3>
    <div
      className="sc-fzoXzr gDXJlj"
    />
  </div>
  <div
    className="sc-fznWqX ldhGcB"
  >
     
    <div
      className="sc-fzoiQi cUsiQd"
    >
      <img
        alt=""
        className="icon"
        onClick={[Function]}
        src="favorites-burger-btn.svg"
      />
    </div>
    <h3
      className="sc-fznJRM fAObgi"
    >
      Favourite
    </h3>
  </div>
  <div
    className="sc-fzqNJr QiOxu"
  >
    <aside>
      <div
        className="sc-fzoXzr gDXJlj"
      />
    </aside>
  </div>
</div>
`;
